### PR TITLE
CCI Tags example

### LINKED ISSUE #199: include overlay caveat in test and details
- [ ] Fix DESC to spell out 'description'
- [ ] Caveat should be above the 'Primary Description' 
- [ ] add others below the primary description
- Suggested Order Text Blocks
- Maintain the 'logic' on `Justification` title replacement for Not Applicable conditions - this should move to the Slot logic?

see: https://www.vuemastery.com/courses/component-design-patterns/slots-fundamentals 

  1. Caveat
  2. Primary Description
  3. Other subs
  4. Check and Fix

So the Slot order should be in both the Results and the Details Tab.
- `caveat`
- `description main`
- `other sub-descriptions.each`

If a caveat is placed in an overlay, such as to make something N/A with an impact of zero:
image

Then place caveat text here, adding a header for the sub-description as we do for 'Justification:'
image

and here, also add a header for each sub-description so that it all flows: e.g. 'Caveat:': \n "text" etc.
image

### LINKED ISSUE #216: Display number of tests in summary
**Is your feature request related to a problem? Please describe.**
Dashboard shows summary at controls level granularity and also estimates percentage compliance level based on these counts.  However, this may not always reflect number of tests run in each control and how close the team is to achieve compliance in each control. For example, a control may be considered as failed even if 1 out of 100 tests in that control failed. 

When we ran nginx baseline @ https://github.com/mitre/nginx-baseline against the ubuntu test environment provided in the inspec training, only 13 out of 42 controls passed and 24 controls failed. If we take a look at tests level, 2,592 tests passed out of 2747. Only 148 tests failed and 7 skipped. This level of granularity in summary dashboard will provide much clearer picture for all stakeholders and reflect the progress made when used in CI/CD or review delta results.

image
 
Numbers by tests included (produced with a simple JSON results parser built to validate the usecase) 
image

**Describe the solution you'd like**
1. It will be beneficial to include tests summary as well, to provide a better picture of the status of tests contributing to overall compliance. Please see the mockup, where summary of tests is also added to provide additional context.
image

2. While number of tests not necessarily mean number of controls and hence compliance, it is the level of granularity that is interesting and useful to developers, testers and security engineers. it will be great to add another row (or separate tab) to display all summary by test cases (instead of summary by controls). This will be immensely helpful to developers, testers and reviewers to quickly look at ONLY failed tests for instance. 

**Describe alternatives you've considered**
Built a command line parser to view this information. Please see above for a screenshot of information produced with such a parser. 

**Desired Impact to Workflow**
Great to visualize the information in different views, as relevant to different team members that are contributing to overall compliance.

### PR DESCRIPTION
**Pull request by aaronlippold**
- [x] as we look at the data counting - https://github.com/mitre/heimdall-lite/issues/216 may be an easy win as well
- [x] did we implament https://github.com/mitre/heimdall-lite/issues/216 ?
  - this may just need to be another PR
  - what do you all think? @Mitriol @cjdoherty 
- [ ] Add the title description for the CCI tag rollover @cjdoherty
  - [ ] make the rollover bring up the description of the CCI as we do for the 800-53 controls
  - [x] add the CCI_Data XML file as we did with the NIST XML data @cjdoherty  
- [x] If possible, see if we can align which CCI is related to which NIST controls @Mitriol @cjdoherty 
- [x] regex/perhaps InSpecJS: NIST desc lookup function and tooltip is having issues with some controls
  - [x] Issues seems to be: a spacing issue between `IA-3(x) vs IA-3 (x)` - look at foundations vs RHEL data @Mitriol @cjdoherty
  - [x] I think the resolution is to make the parser greedy or kill all spaces in the `keys` of the lookup. 
- [x] Change column title to `NIST SP 800-53 Security Controls & CCIs` @aaronlippold 

Signed-off-by: Aaron Lippold <lippold@gmail.com>

### CODE REVIEW

**Review by Mitriol — APPROVED** [review]

File: src/components/cards/controltable/ControlRowHeader.vue [hunk 1]
```diff
@@ -140,8 +144,10 @@ export default class ControlRowHeader extends ControlRowHeaderProps {
 
   get filteredNistTags(): string[] {
     {
-      var value = "Rev_4";
-      return this._control.hdf.raw_nist_tags.filter(item => item !== value);
+      let ignored = /Rev_\d/;
```
Comment (aaronlippold) [inline src/components/cards/controltable/ControlRowHeader.vue]: This doesn't seem to be removing the Rev_4 item from the array

**Comment by aaronlippold** [comment]
We need to do a rebase of these PRs to keep things clean

**Comment by aaronlippold** [comment]
@cjbuchanan @Mitriol please note the attached .zip file and npm package that was pushed as an example. We may be able to borrow from there?

File: tests/scripts/make-all.sh.broken [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+
```
Comment (rbclark) [inline tests/scripts/make-all.sh.broken:3]: This file seems unnecessary as part of the PR, or accidental.

File: src/components/cards/controltable/ControlRowHeader.vue [hunk 3]
```diff
@@ -137,6 +154,17 @@ export default class ControlRowHeader extends ControlRowHeaderProps {
...
+
+  get fixedCCI(): string[] {
+    let cci: string | string[] | undefined = this._control.hdf.wraps.tags.cci;
+    if (!cci) {
+      return [];
+    } else if (Array.isArray(cci)) {
+      return cci;
+    } else {
+      return [cci];
+    }
```
Comment (rbclark) [inline src/components/cards/controltable/ControlRowHeader.vue:166]: This function can probably be replaced with 

```
return [].concat(this._control.hdf.wraps.tags.cci || []);
```

File: tests/backups/info-logs/test.log [hunk 4]
```diff
@@ -0,0 +1,276 @@
+../raw_data/71859-71861-71863.json
```
Comment (rbclark) [inline tests/backups/info-logs/test.log:1]: Was this log file intentionally included? It looks like an attempt was made to ignore log files above.

File: tests/scripts/make-info.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+do
+case "${option}"
+in
+p) INFO_PATH=${OPTARG:-.};;
+l) LOG=${OPTARG:-$(date)};;
+o) OUTPUT=${OPTARG:-.}
+esac
+done
+
+ls $INFO_PATH/*.json | sort | xargs -I $$$ sh -c "echo $$$ ; inspec_tools summary -j $$$ -o $$$.info; test -f $$$.info ; echo .info created for $$$ ; continue" | tee $LOG.log
```
Comment (rbclark) [inline tests/scripts/make-info.sh:13]: These files are placed in the `tests/` folder but I don't see any automated testing being done that is using these files.

File: tests/scripts/make-counts.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+while getopts p:l: option
+do
+case "${option}"
+in
+p) INFO_PATH=${OPTARG:-.};;
+l) LOG=${OPTARG:-$(date)};;
+esac
+done
+
+ls $INFO_PATH/*.info | xargs -I {} sh -c "cat {} | jq .status > {}.counts" | tee $LOG.log
```
Comment (rbclark) [inline tests/scripts/make-counts.sh:12]: These files are placed in the `tests/` folder but I don't see any automated testing being done that is using these files.

**Review by rbclark — CHANGES_REQUESTED** [review]
Additionally, there are 2 binary files included in this PR that don't seem to be used anywhere, `heimdall-logo.sketch` and a large 9MB `old-info.tar.gz` which if merged would add 9MB to every download of the repo.

**Comment by cjdoherty** [comment]
This is the wrong branch, the new one is cci-tooltips-final

**Pull request closed without merging**